src/core/folder.h

```diff
@@ -179,6 +183,8 @@ private Q_SLOTS:
     bool defer_content_test : 1;
 
     static std::unordered_map<FilePath, std::weak_ptr<Folder>, FilePathHash> cache_;
+    static FilePath cutFilesDirPath_;
+    static std::shared_ptr<const HashSet> cutFilesHashSet_;
```

Comment:
For the _desktop-icons-not-showing-cut-status_ bug, I confirm it. I don't use desktop icons, so it didn't come up to me to test this use case.